tests/test_widgets.py

```diff
@@ -1059,6 +1059,59 @@ def f(x: Set[Lit]): ...
     assert sel.choices == get_args(Lit)
 
 
+def test_separator_singleton():
+    from magicgui.types import Separator, _Separator
+
+    sep1 = Separator
+    sep2 = _Separator()
+    assert sep1 is sep2
+
+
+def test_separator(backend: str):
+    from magicgui.types import Separator
+
+    use_app(backend)
+
+    sep = [[Separator] * (i + 1) for i in range(4)]
+    a = [1, 2, *sep[0], 4, *sep[1], 6, 7, *sep[2], 9, *sep[3], 11, 12, *sep[0], 14, *sep[0]]
+    b = [1, 2, *sep[0], 4, *sep[1], 6, 7, *sep[2], 9, *sep[3], 6, 7, *sep[0], 9, *sep[0]]
+    b2 = [1, 2, *sep[0], 4, *sep[1], 6, 7, *sep[2], 9, *sep[3], *sep[0], *sep[0]]
+
+    @magicgui(
+        call_button="Test Separator",
+        a={"widget_type": "ComboBox", "choices": a},
+        b={"widget_type": "ComboBox", "choices": b},
+        result_widget=True,
+    )
+    def widget_with_separator(a=a[0], b=b[0]):
+        return
```

Comment:
no need to bring `@magicgui` into this :) let's just make comboboxes:

```python
    combo_a = widgets.ComboBox(choices=a, value=a[0])
    combo_b = widgets.ComboBox(choices=b, value=b[0])
```

and use them below `widget_with_separator.a` -> `combo_a`